prefect_gcp/workers/cloud_run_v2.py

```diff
@@ -0,0 +1,838 @@
+import re
+import shlex
+import time
+from typing import TYPE_CHECKING, Any, Dict, List, Literal, Optional
+
+from anyio.abc import TaskStatus
+from google.api_core.client_options import ClientOptions
+from googleapiclient import discovery
+
+# noinspection PyProtectedMember
+from googleapiclient.discovery import Resource
+from googleapiclient.errors import HttpError
+from prefect.exceptions import InfrastructureNotFound
+from prefect.logging.loggers import PrefectLogAdapter
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+from prefect.utilities.dockerutils import get_prefect_image_name
+from prefect.utilities.pydantic import JsonPatch
+from prefect.workers.base import (
+    BaseJobConfiguration,
+    BaseVariables,
+    BaseWorker,
+    BaseWorkerResult,
+)
+from pydantic import VERSION as PYDANTIC_VERSION
+
+if PYDANTIC_VERSION.startswith("2."):
+    from pydantic.v1 import Field, validator
+else:
+    from pydantic import Field, validator
+
+from prefect_gcp.cloud_run_v2 import CloudRunJobV2Result, ExecutionV2, JobV2
+from prefect_gcp.credentials import GcpCredentials
+
+if TYPE_CHECKING:
+    from prefect.client.schemas import FlowRun
+    from prefect.server.schemas.core import Flow
+    from prefect.server.schemas.responses import DeploymentResponse
+
+
+def _get_default_job_body_template() -> Dict[str, Any]:
+    """
+    Returns the default job body template for the Cloud Run worker.
+
+    Returns:
+        The default job body template.
+    """
+    return {
+        "client": "prefect",
+        "launchStage": "{{ launch_stage }}",
+        "template": {
+            "template": {
+                "maxRetries": "{{ max_retries }}",
+                "timeout": "{{ timeout }}",
+                "containers": [
+                    {
+                        "env": [],
+                        "image": "{{ image }}",
+                        "command": "{{ command }}",
+                        "args": "{{ args }}",
+                        "resources": {
+                            "limits": {
+                                "cpu": "{{ cpu }}",
+                                "memory": "{{ memory }}",
+                            },
+                        },
+                    },
+                ],
+            }
+        },
+    }
+
+
+def _get_base_job_body() -> Dict[str, Any]:
+    """
+    Returns the base job body for the Cloud Run worker's job body validation.
+
+    Returns:
+        The base job body.
+    """
+    return {
+        "template": {
+            "template": {
+                "containers": [],
+            },
+        },
+    }
+
+
+class CloudRunWorkerJobV2Configuration(BaseJobConfiguration):
+    credentials: Optional[GcpCredentials] = Field(
```

Comment:
[`interrogate`](https://github.com/econchick/interrogate) is mad because you don't have a docstring, not because of unit test coverage 🙂

Reply:
Fail on my part lol, thanks for catching that 😸 !